oss_src/process/process_unix.cpp

```diff
@@ -119,11 +119,14 @@ bool process::popen(const std::string &cmd,
     return false;
   } else if(pid == 0) {
     //***In child***
-    if(child_write_fd > -1) {
-      dup2(write_fd, child_write_fd);
-    }
-    close(write_fd);
     close(read_fd);
+    if((child_write_fd > -1) && (write_fd != child_write_fd)) {
+      errno = 0;
+      if(dup2(write_fd, child_write_fd) != child_write_fd) {
+        logstream(LOG_ERROR) << "dup2 failed to duplicate fd!" << strerror(errno) << std::endl;
```

Comment:
I might just assert(dup2(write_fd ....) != child_write_fd).

(the c assert as opposed to our ASSERT)